test/functional/checkin.js

```diff
@@ -0,0 +1,113 @@
+import { assert } from 'chai';
+import { addAction } from '../util/database';
+import { post } from '../util/mock-client';
+import server from '../../.dist/lib/server';
+import CheckinStore from '../../.dist/store/checkin-store';
+import ItemStore from '../../.dist/store/item-store';
+import ModelStore from '../../.dist/store/model-store';
+
+
+describe('Chekin items', () => {
+
+    before(() => {
+        return server.start(8080);
+    });
+
+    after(() => {
+        return server.stop();
+    });
+
+    beforeEach(() => {
+        return addAction('CLEAR_ALL_DATA').then(() => {
+            return Promise.all([
+                addAction('NEW_MODEL', {
+                    name: 'Widget',
+                    description: 'A widget',
+                    manufacturer: 'The Factory',
+                    vendor: 'The Store',
+                    location: 'The shelf',
+                    isFaulty: false,
+                    faultDescription: '',
+                    price: 3.50,
+                    count: 10
+                }),
+                addAction('NEW_MODEL', {
+                    name: 'OtherThing',
+                    description: 'Not a widget',
+                    manufacturer: 'The Factory',
+                    vendor: 'The Store',
+                    location: 'The shelf',
+                    isFaulty: false,
+                    faultDescription: '',
+                    price: 3.50,
+                    count: 10
+                }),
+                addAction('NEW_STUDENT', {
+                    id: 123456,
+                    name: 'John von Neumann',
+                    email: 'jvn@example.com',
+                    major: 'Checmical Engineering & Mathematics',
+                    status: 'C - Current'
+                })
+            ]).then(() => {
+                let modAddr1 = ModelStore.getModels()[0].address;
+                let modAddr2 = ModelStore.getModels()[1].address;
+                return Promise.all([
+                    addAction('NEW_ITEM', {
+                        modelAddress: modAddr1
+                    }),
+                    addAction('NEW_ITEM', {
+                        modelAddress: modAddr1
+                    }),
+                    addAction('NEW_ITEM', {
+                        modelAddress: modAddr2
+                    })
+                ]);
+            }).then(() => {
+                return addAction('NEW_CHECKOUT', {
+                    itemAddresses: ItemStore.getItems().map(item => item.address),
+                    studentId: '123456'
+                });
+            });
+        });
+    });
+
+    it('should checkin an item', () => {
+        assert.lengthOf(CheckinStore.getCheckins(), 0);
+        assert.lengthOf(ItemStore.getItems().filter(item => item.status === 'CHECKED_OUT'), 3);
+
+        return post('checkin', {
+            studentId: '123456',
+            itemAddress: ItemStore.getItems()[0].address
+        }).then(() => {
```

Comment:
(GitHub got confused with this big diff; this is no longer in reference to the [correct line](https://github.com/TheFourFifths/consus/pull/52/commits/1b373a62cd385b199bde5a21ba3f4222868edbd7#diff-3dd376d611ae34cb10e2c491336e5f26R82).)